config/regex.js

```diff
@@ -55,15 +55,17 @@ module.exports = {
     QUALIFIED_STAGE_NAME: /^stage@([\w-]+)$/,
     // Stage trigger like ~stage@deploy or ~stage@stageName:jobName or stage@deploy or stage@stageName:jobName
     STAGE_TRIGGER: /^~?stage@([\w-]+)(?::([\w-]+))?$/,
+    // Stage trigger like ~sd@26:stage@alpha:setup, sd@26:stage@alpha:deploy
+    EXTERNAL_STAGE_TRIGGER: /^~?sd@\d+:(([\w-]+)|(stage@([\w-]+):([\w-]+)))$/,
     // Don't combine EXTERNAL_TRIGGER and EXTERNAL_TRIGGER_AND for backward compatibility
     // BlockBy does not support EXTERNAL_TRIGGER_AND
-    // External trigger like ~sd@123:component (OR case)
-    EXTERNAL_TRIGGER: /^~sd@(\d+):([\w-]+)$/,
-    // External trigger like sd@123:component (AND case)
-    EXTERNAL_TRIGGER_AND: /^sd@(\d+):([\w-]+)$/,
+    // External trigger like ~sd@123:component, ~sd@26:stage@alpha:setup (OR case)
+    EXTERNAL_TRIGGER: /^~sd@(\d+):(([\w-]+)|(stage@([\w-]+):setup))$/,
```

Comment:
```suggestion
    EXTERNAL_TRIGGER: /^~sd@(\d+):(?:([\w-]+)|(stage@([\w-]+):setup))$/,
```
Before:
<img width="555" alt="Screenshot 2024-08-02 at 9 37 43 AM" src="https://github.com/user-attachments/assets/d761d38f-b104-4bec-9ee4-880aefe865f1">


After:
<img width="391" alt="Screenshot 2024-08-02 at 9 37 29 AM" src="https://github.com/user-attachments/assets/f87ed231-e2cf-416d-a7b0-cf8ed864a960">